### ISSUE TITLE
iOS: Save stories offline

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1111]
Allow saving of stories while offline, as well as unsaying stories that were saved while offline. No need to support unsaying of stories while offline if they weren't saved in the same session. Re-use the mark-read temp table currently used for a similar process.

**Comment by Dejal** [comment]
Looking at this one, when marking a bunch of stories as read, there's an `reader/mark_story_hashes_as_read` endpoint that takes an array of story hashes.  But for marking as saved, there is only `reader/mark_story_as_starred`, that marks a single story. If not too difficult, can you please add a `reader/mark_story_hashes_as_starred` endpoint that is equivalent to the `mark_story_hashes_as_read` one, but for starred/saved, to avoid having to do multiple network requests? Thanks.

**Comment by samuelclay** [comment]
People don't save that many stories. Just perform them sequentially. It's OK if the progress bar goes 1 by 1 by 1.

**Comment by Dejal** [comment]
Fair enough.  However, I might be stuck: given a feed ID and story hash (saved in the database while offline), how can I get the story dictionary when syncing once back online? I don't see anywhere that does that, when the story isn't necessarily in `storiesCollection.activeFeedStories`. Maybe I need to save more info in the database, or something.

**Comment by samuelclay** [comment]
You'll need to store that information when the user saves the story. Make a new table (follow the layout for the read stories table).

**Comment by Dejal** [comment]
Okay. I already have the new table, so will add the extra info. Note, I'll continue with this next week.
